File: Practice/33/Python/33.py
out = print

def create(ln, x0 = 0, c = 0):
    arr = []
    arr.append(x0)
    for i in range(1, ln):
        arr.append(arr[i-1] + c)
    return arr

def sort(arr):
    for i in range(1, len(arr)):
        a = arr[i]
        b = i-1
        while b >=0 and a < arr[b]:
                arr[b+1] = arr[b]
                b -= 1
        arr[b+1] = a
    return arr

def print(arr):
    out('[', end='')
    for i in range(len(arr) - 1):
        out(arr[i], end=', ')
    out(arr[len(arr) - 1], ']', sep='')
    return arr

out('Введите длину массива, и два числа: начальный член и шаг арифметической прогрессии')
ln = int(input())
x0 = int(input())
c = int(input())
arr = create(ln, x0, c)
sort(arr)
print(arr)

File: Practice/33/C++/33/33.cpp
#include <iostream>

using namespace std;

int* create(int len, int x0, int c);
int* sort(int* arr, int len);
int* print(int* arr, int len);

int main()
{
    int len, x0, c;
    x0 = 0;
    c = 0;
    setlocale(LC_ALL, "rus");
    cout << "Введите длину массива, и два числа: начальный член и шаг арифметической прогрессии" << endl;
    cin >> len >> x0 >> c;
    int *arr = create(len, x0, c);
    sort(arr, len);
    print(arr, len);
    delete[] arr;
}

int* create(int len, int x0, int c)
{
    int* arr = new int[len];
    arr[0] = x0;
    for (int i = 1; i < len; i++)
    {
        arr[i] = arr[i - 1] + c;
    }
    return arr;
}

int* sort(int* arr, int len)
{
    int a, b;
    for (int i = 1; i < len; i++)
    {
        a = arr[i];
        b = i;
        while ((b > 0) and (arr[b - 1] > a))
        {
            arr[b] = arr[b - 1];
            b--;
        }
        arr[b] = a;
    }
    return arr;
}

int* print(int* arr, int len)
{
    cout << "[" << arr[0];
    for (int i = 1; i < len; i++)
    {
        cout << ", " << arr[i];
    }
    cout << "]";
    return arr;
}